modules/git/repo_commit_nogogit.go

```diff
@@ -123,6 +123,18 @@ func (repo *Repository) getCommitFromBatchReader(rd *bufio.Reader, id SHA1) (*Co
 		}
 
 		return commit, nil
+	case "blob":
+		fallthrough
+	case "tree":
+		log.Debug("Special typ: %s", typ)
```

Comment:
I don't think this log debug line will be useful. I think it can be dropped.

Reply:
You're going to need to add this/handle this in the gogit variant code too

Reply:
Agree. I will remove it.

Reply:
> You're going to need to add this/handle this in the gogit variant code too

Well, you are right. I do a refactor for the getCommit() of gogit variant code in the latest patch.